src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -116,7 +116,8 @@ protected List<ImportFilesResultItemViewModel> call() {
 
                     try {
                         if (FileUtil.isPDFFile(file)) {
-                            var pdfImporterResult = contentImporter.importPDFContent(file, bibDatabaseContext.getMetaData());
+                            Optional<Path> globalFileDir = bibDatabaseContext.getFirstExistingFileDir(preferencesService.getFilePreferences());
+                            var pdfImporterResult = contentImporter.importPDFContent(file, globalFileDir);
```

Comment:
Optionals should not be passed as parameter, they should only be used for return types. Better

Path workingDirectory = databaseContext.getFirstExistingFileDir(filePreferences)
                                               .orElse(filePreferences.getWorkingDirectory());

Reply:
Yes, I agree. This feels like a code smell. I've pushed 2 new commits: the 1st removes the `Optional` and the 2nd one is about renaming the `Path` variable.